packages/realm/src/Configuration.ts

```diff
@@ -243,6 +243,7 @@ export function validateConfiguration(config: unknown): asserts config is Config
   }
   if (sync !== undefined) {
     assert(!onMigration, "The realm configuration options 'onMigration' and 'sync' cannot both be defined.");
+    assert(!inMemory, "The realm configuration options 'inMemory' and 'sync' cannot both be defined.");
```

Comment:
We need to check for `undefined` here since `inMemory` is a boolean. (The above `!onMigration` works because it's a function.)
```suggestion
    assert(inMemory === undefined, "The realm configuration options 'inMemory' and 'sync' cannot both be defined.");
```